apps/ledger-live-desktop/src/internal/transportHandler.ts

```diff
@@ -0,0 +1,108 @@
+import { Subject } from "rxjs";
+import { withDevice } from "@ledgerhq/live-common/hw/deviceAccess";
+
+type APDUMessage = { apduHex: string; requestId: string };
+const transports = new Map<string, Subject<APDUMessage>>();
+
+export const transportOpenChannel = "transport:open";
+export const transportExchangeChannel = "transport:exchange";
+export const transportCloseChannel = "transport:close";
+
+export const transportOpen = ({
+  data,
+  requestId,
+}: {
+  data: { descriptor: string };
+  requestId: string;
+}) => {
+  const subjectExist = transports.get(data.descriptor);
+  if (subjectExist) {
+    // Should we return saying the transport is open ?
```

Comment:
probably not. it indicates a logic problem in our app.

I wonder therefore if there are a better `Error` to use so we can more clearly identify and remap it to a "worded error".